extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -531,85 +566,76 @@ class LifecycleTask extends BackbeatTask {
         });
     }
 
-    _compareContents(bucketData, lcRules, contents, log, done) {
-        if (!contents) {
+    _compareList(bucketData, lcRules, contents, log, versioned, done) {
+        if (!contents.length) {
             return done();
         }
         return async.eachLimit(contents, CONCURRENCY_DEFAULT, (obj, cb) => {
             async.waterfall([
                 next => this._getRules(bucketData, lcRules, obj, log, next),
-                (applicableRules, next) => this._compareObject(
-                    bucketData, obj, applicableRules, log, next),
+                (applicableRules, next) => {
+                    if (versioned) {
+                        return this._compareAVersion(bucketData, obj,
+                            applicableRules, log, next);
+                    }
+                    return this._compareAnObject(bucketData, obj,
+                        applicableRules, log, next);
+                },
             ], cb);
         }, done);
     }
 
-    _compareVersions(bucketData, lcRules, versions, log, done) {
-        if (!versions) {
+    /**
+     * Compare a version to most applicable rules
+     * @param {object} bucketData - bucket data
+     * @param {object} version - single version from `_getObjectVersions`
+     * @param {object} rules - most applicable rules from `_getApplicableRules`
+     * @param {Logger.newRequestLogger} log - logger object
+     * @param {function} done - callback(error, data)
+     * @return {undefined}
+     */
+    _compareAVersion(bucketData, version, rules, log, done) {
```

Comment:
same, you can rename instead to `_compareVersion` (though I like the idea of separating a single version processing in its own function).